src/clusterfuzz/_internal/bot/tasks/utasks/minimize_task.py

```diff
@@ -586,23 +579,57 @@ def utask_main(uworker_input):
     testcase.put()
 
     if check_deadline_exceeded_and_store_partial_minimized_testcase(
-        deadline, uworker_input.testcase_id, uworker_input.job_type,
-        input_directory, file_list, data, testcase_file_path):
-      return
+        deadline, testcase_id, uworker_input.job_type, input_directory,
+        file_list, data, testcase_file_path):
+      return None
 
   command = testcase_manager.get_command_line_for_application(
       testcase_file_path, app_args=app_arguments, needs_http=testcase.http_flag)
   last_crash_result = test_runner.last_failing_result
 
-  store_minimized_testcase(testcase, input_directory, file_list, data,
-                           testcase_file_path)
-  finalize_testcase(
-      uworker_input.testcase_id,
-      command,
-      last_crash_result,
+  # TODO(pgrace): This is a temporary bookmark comment
+
+  return uworker_io.UworkerOutput(
+      testcase=testcase,
+      input_directory=input_directory,
+      file_list=file_list,
+      data=data,
+      testcase_file_path=testcase_file_path,
+      command=command,
+      last_crash_result=last_crash_result,
       flaky_stack=flaky_stack)
 
 
+def utask_postprocess(output):
+  store_minimized_testcase(output.testcase, output.input_directory,
+                           output.file_list, output.data,
+                           output.testcase_file_path)
+  finalize_testcase(
+      output.testcase,
+      output.command,
+      output.last_crash_result,
+      flaky_stack=output.flaky_stack)
+  del output
+
+
+def handle_setup_failure(output):
+  """Handles errors occuring during setup."""
+  logs.log_error('Unable to setup build for minimization.')
```

Comment:
Error handling is a little hard in this project. You also need to add this boilerplate to utaks_postprocess:
 https://github.com/google/clusterfuzz/blob/961b23b21fd522b19eb66de9d5f3bf4d47b7b4be/src/clusterfuzz/_internal/bot/tasks/utasks/analyze_task.py#L408C1-L422C11

And register it in uworker_handle_errors:
https://github.com/google/clusterfuzz/blob/master/src/clusterfuzz/_internal/bot/tasks/utasks/uworker_handle_errors.py#L37

I wonder if this is a bad design, since it sort of forces a circular dependency

Reply:
ooh done! added the boilerplate